sqlglot/optimizer/qualify_columns.py

```diff
@@ -380,18 +410,89 @@ def _qualify_columns(scope: Scope, resolver: Resolver) -> None:
                     column.set("table", column_table)
 
 
+def _expand_struct_stars(
+    expression: exp.Dot,
+    next_alias_name: t.Callable[[], str],
+) -> t.List[exp.Alias]:
+    struct_fields: t.List[str] = []
+
+    # [BigQuery] Expand/Flatten foo.bar.* where bar is a struct column
+    starting_col = expression.find(exp.Column)
+    assert starting_col
+
+    dot_parts = expression.parts
+    table_name = starting_col.table
+
+    # If there's more nested levels specified, start the DFS traversal from the nested struct first
+    for part in dot_parts[2:-1]:
+        datatype = starting_col.find(exp.DataType) or (
+            starting_col.type and starting_col.type.find(exp.DataType)
+        )
+        assert datatype
+
+        for expr in datatype.expressions:
+            if not isinstance(expr.this, exp.Identifier):
+                return []
+
+            if expr.this.this == part.this:
+                starting_col = expr
+                assert starting_col
+
+    if not starting_col or starting_col.this.this != dot_parts[-2].this:
+        raise OptimizeError(f"Unknown struct: {part.this}")
+
+    starting_name = ".".join([part.this for part in dot_parts[1:-2]])
+    stack = [(starting_col, starting_name)]
+
+    # Traverse & flatten the starting_col struct
+    while stack:
+        col, path_name = stack.pop()
+        assert col
+
+        if not isinstance(col.this, exp.Identifier):
+            return []
+
+        path_name = col.this.this if not path_name else f"{path_name}.{col.this.this}"
+        datatype = col.find(exp.DataType) or (col.type and col.type.find(exp.DataType))
+        assert datatype
+
+        # Collect the non-struct columns
+        if datatype.this != exp.DataType.Type.STRUCT:
+            struct_fields.append(path_name)
+        else:
+            for expr in reversed(datatype.expressions):
+                stack.append((expr, path_name))
+
+    new_selections = []
+
+    for struct_field in struct_fields:
+        root, *parts = struct_field.split(".")
+        alias_ = next_alias_name()
+        new_column: exp.Column | exp.Dot = exp.column(root, table=table_name)
+
+        if parts:
+            new_column = exp.Dot.build([new_column, *[exp.Identifier(this=part) for part in parts]])
```

Comment:
Here, `parts` is produced by splitting a string on `.`. Then, we use it to produce a new `Dot` where we create identifiers from each part. What happens if there are quotes? If a `part` is quoted, it seems that you'll add `"name"` to `Identifier.this` which is incorrect.

We need to be very careful with quotes because they will mess things up eg. in SQLMesh if not done right. Let's test this path as well.